Update packages, add localisation, add search

EDIflyer opened the pull request:
Apologies @JaxonWright - I know this is a big of a mega PR but I got into a bit of a run yesterday evening!

I've built and deployed it to my own server and it seems to work well on my Model 3 but appreciate needs checking.

Fixes #51 (add localisation)
Fixes #6 (add search)

At a high level these were the main changes made:
- updated dependencies
- addition of a Dockerfile to allow an easy build for deployment - I've also written a deployment script I use on my server but didn't want to add that to the repo in case it was unwanted/too localised to me (e.g., I use Nginx Proxy Manager), have attached instead! - [testube.txt](https://github.com/user-attachments/files/21707672/testube.txt) (just change extension to `.sh` and `chmod +x` to run)
- added a search box (including option to include hidden services)
- added localisation - I did this by allowing services to be noted as `Worldwide`, `US` or `UK` - I went through and allocated each one, hopefully correctly! It should be easily extensible for other countries. I also removed the reset option and instead offered select all/select none for each region (as well as an added an unfiltered option to show all services). Switching between regions is possible when selecting services, changes only happen when you hit save
- added UK services
- updated/added a few images for services 
- fixed the font colour on the about page
- added a 'no' option for the full screen popup
- added ability in settings to remove horizontal scrolling as it seems a shame to have that if you've lots of one service but not many of others, so they now just wrap if you switch that off
- added ability to define a different target URL for when heading off to YouTube to go fullscreen (to allow for local deployment)

_A more detailed Copilot summary is as follows:_
This pull request introduces several major improvements to the Angular application's infrastructure, user experience, and dependency management. The most significant changes include the addition of Docker support for building and serving the app with nginx, a comprehensive redesign of the settings dialog with new region and layout preferences, and an upgrade of Angular and related dependencies to version 20. These updates enhance deployment flexibility, user customization, and overall maintainability.

**Infrastructure & Deployment**

* Added a multi-stage `Dockerfile` for building the Angular app and serving it with nginx, including a custom `nginx.conf` for optimized routing, caching, and security headers. (`Dockerfile`, `nginx.conf`) [[1]](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R1-R35) [[2]](diffhunk://#diff-1f91f64bfa3fb0a1ece881887af0a2356b8c529a96fb91c1894745b2a1a009aaR1-R47)
* Created a `.dockerignore` file to exclude unnecessary files and directories from Docker builds, improving build performance and security. (`.dockerignore`)

**User Experience: Settings Dialog**

* Redesigned the settings dialog to use tabs for "Services" and "Preferences," added region selection with flag icons, improved source visibility controls, and enabled layout and fullscreen URL customization. (`settings-dialog.component.html`, `settings-dialog.component.scss`) [[1]](diffhunk://#diff-24711d899da6b745263b7101dba017666fe4d54b6e3a7198cc629809b51cd8d1R1-L21) [[2]](diffhunk://#diff-364d96613478d358e979150113f91c8147aeac2dc279803d660edd62a9c3d730R1-R64) [[3]](diffhunk://#diff-364d96613478d358e979150113f91c8147aeac2dc279803d660edd62a9c3d730L10-R243)

**Dependency Management**

* Upgraded Angular and related dependencies to version 20, and updated other libraries to their latest versions for improved performance, security, and compatibility. (`package.json`) [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L15-R42) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L53-R53)

**Angular Project Configuration**

* Added schematics configuration to `angular.json` for more consistent component, directive, and service generation, improving developer experience. (`angular.json`)

**Component & Module Enhancements**

* Improved the fullscreen prompt in `AppComponent` to use a customizable URL from the settings, and added support for `MatSelectModule` in the app module for the new region selection UI. (`app.component.ts`, `app.module.ts`) [[1]](diffhunk://#diff-884f7f49640e5923f6bcac4c51d90340330a178f662defbe61e5f5aac1c512deL2-R3) [[2]](diffhunk://#diff-884f7f49640e5923f6bcac4c51d90340330a178f662defbe61e5f5aac1c512deL16-R57) [[3]](diffhunk://#diff-502d295b435b10ba99f5377504ca51149312e3cbebbbaa78ac5d925cb04cbfb0R30) [[4]](diffhunk://#diff-502d295b435b10ba99f5377504ca51149312e3cbebbbaa78ac5d925cb04cbfb0R54)

EDIflyer committed:
feat: add Dockerfile

EDIflyer committed:
update: Angular CLI and core packages to v20.x

EDIflyer committed:
update: upgrade other packages to latest versions

EDIflyer committed:
fix: ensure TypeScript compatibility with Angular 20.x

EDIflyer committed:
update: final minor version updates for remaining packages

EDIflyer committed:
fix: update Dockerfile for Angular application with proper multi-stage build

EDIflyer committed:
feat: add regions to services

EDIflyer committed:
feat: update application settings to have regional filtering for services

EDIflyer committed:
feat: add UK streaming services

EDIflyer committed:
feat: add option to disable horizontal scrolling

EDIflyer committed:
feat: add option for different fullscreen URL

EDIflyer committed:
fix: styling for setting and fullscreen popup

EDIflyer committed:
feat: add search functionality (including for hidden services)

EDIflyer committed:
fix: update britbox logo

EDIflyer committed:
fix: ensure checkbox selectable on mobile in sources list

EDIflyer commented:
Haha yep sorry, I know it's a lot! No rush at all. I mainly added Docker for those wishing to self host just to make it easier (I'm running it via that method at https://testube.ediflyer.net), but could have that and the deployment script just in the documentation if preferred.

JaxonWright commented:
> I don't quite follow? Picking a different region filters the list of services, you can then pick select all/select none to bulk change, and/or click individual ones.

Sure, it filters the list in the dialog, but I could never get it to apply to the actual data on the page. Yes, I clicked Save.

EDIflyer committed:
fix: deal with localstorage not always remembering URL change

EDIflyer committed:
fix: move search to left, fix padding, add close for Custom URL

EDIflyer committed:
feat: update default fullscreen URL to currently hosted one to cope with deployment to different servers

EDIflyer committed:
fix: ensure search box positioned correctly at different screen resolutions

EDIflyer committed:
feat: allow sorting of services by popularity or alphabetical order

EDIflyer committed:
fix: tweak setting scroll

EDIflyer commented:
Sorry for the delay looking at this - been kept busy with work!

I've now made the following updates:
- services back to being sorted by your previous popularity order by default, however now an option in prefs to switch to alphabetical
- search box moved to left
- I've added an X option for custom URL so this can be closed again, search box moves out OK which this expanded too
- I've updated the default URL when opening fullscreen to be the one it is hosted on (as that makes more sense for custom deployments), however users can still manually change this if they wish

I've redeployed to my personal server and it's working perfectly re filtering, so I'm afraid I can't replicate that.  Example of it working fine for me:
![Animation](https://github.com/user-attachments/assets/058b7c6f-ae56-4540-aeb4-5e5a87bba063)
